### PR TITLE
Add support for handling notifications in the background

### LINKED ISSUE #7: Support for silent notifications
Reading through the Apple documentation, I came across silent notifications (basically background app refresh). When the APS payload contains `content-available` key with a value of `1` it calls `application:didReceiveRemoteNotification:fetchCompletionHandler:`. I think for this library to be considered complete, it needs to support this delegate method. It's not clear to me if `content-available` can be combined with the others, or if setting `content-available` to `1` makes it a silent notification.

https://developer.apple.com/library/ios/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/Chapters/TheNotificationPayload.html#//apple_ref/doc/uid/TP40008194-CH107-SW1

### PR DESCRIPTION
**Pull request by andrewhavens**
This PR adds support for handling push notifications in the background. This PR fixes #7.

### CODE REVIEW

**Comment by jamonholmgren** [comment]
You probably need `osx_image: xcode7.2` in the travis yml file.

**Comment by kevinvangelder** [comment]
That's cool! Style is up to Jamon, but I would move the `valid_responses` on line 98 into a protected method similar to `map_notification_symbol`.

**Comment by markrickert** [comment]
:+1:

File: lib/ProMotion/delegate_notifications.rb [hunk 1]
```diff
@@ -110,5 +115,18 @@ def map_notification_symbol(symbol)
       }[symbol] || UIRemoteNotificationTypeNone
     end
 
+    def background_fetch_result(result)
```
Comment (andrewhavens) [inline lib/ProMotion/delegate_notifications.rb]: @kevinvangelder How about this?

Comment (kevinvangelder) [inline lib/ProMotion/delegate_notifications.rb]: @andrewhavens I like that better.

**Comment by dchersey** [comment]
Pinging this PR -- can we merge this in?

**Comment by andrewhavens** [comment]
I have merged this PR, but will probably need to wait for @jamonholmgren to release a new gem version. Meanwhile, @dchersey you can set your Gemfile to reference the GitHub master branch.

**Comment by dchersey** [comment]
Awesome, thanks!

**Pull request merged**